Update the CRD descriptions

dhaiducek opened the pull request:
ref: https://issues.redhat.com/browse/ACM-8992

yiraeChristineKim commented on api/v1/certificatepolicy_types.go:
```diff
@@ -69,73 +66,120 @@ func (t Target) String() string {
-	// 'matchExpressions' and 'matchLabels' are both not provided, 'include' must be provided to
+
+	// NamespaceSelector defines the list of namespaces to include/exclude for objects defined in
+	// spec.objectTemplates. All selector rules are ANDed. If Include is not provided but
```
What is ANDed

dhaiducek replied:
As in all results are combined using a logical "AND" operation.

dhaiducek commented:
/hold for reviews

yiraeChristineKim reviewed (approved):
To me it is good

dockerymick commented:
Will review soon, finishing up doc tasks

dhaiducek commented on api/v1/certificatepolicy_types.go:
```diff
@@ -69,73 +66,121 @@ func (t Target) String() string {
 	Severity string `json:"severity,omitempty"`
-	// Minimum duration before a certificate expires that it is considered non-compliant. Golang's time units only.
+
+	// MinDuration is the minimum duration for a certificate's expiration where any shorter duration
+	// is considered non-compliant. The value follows the Golang duration format 0h0m0s where hours
+	// (h) is the greatest accepted unit of time.
```
Updated!

Right, the units are a Golang thing:
```
// There is no definition for units of Day or larger
// to avoid confusion across daylight savings time zone transitions.
```
https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/time/time.go;l=625

mprahl commented on api/v1/certificatepolicy_types.go:
```diff
@@ -69,73 +66,121 @@ func (t Target) String() string {
-	// Minimum duration before a certificate expires that it is considered non-compliant. Golang's time units only.
+
+	// MinDuration is the minimum duration for a certificate's expiration where any shorter duration
+	// is considered non-compliant. The value follows the Golang duration format 0h0m0s where hours
```
I think we use the spelling `noncompliant` most places. What do you think about using that consistently?

dhaiducek replied:
Yes, I'll update it!

mprahl commented on api/v1/certificatepolicy_types.go:
```diff
@@ -69,73 +66,121 @@ func (t Target) String() string {
 
-// CertificatePolicy is the Schema for the certificatepolicies API
+// CertificatePolicy is the Schema for the certificatepolicies API. Certificate policy monitors
+// certificates on a cluster against user-defined restrictions and returns non-compliance if any
```
```suggestion
// certificates on a cluster against user-defined restrictions and returns noncompliant if any
```

dockerymick commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -37,71 +42,80 @@ spec:
-                  spec.objectTemplates. All selector rules are ANDed. If 'include' is not provided but
-                  'matchLabels' and/or 'matchExpressions' are, 'include' will behave as if ['*'] were given. If
-                  'matchExpressions' and 'matchLabels' are both not provided, 'include' must be provided to
+                  NamespaceSelector defines the list of namespaces to include/exclude for objects defined in
+                  spec.objectTemplates. All selector rules are ANDed. If Include is not provided but
+                  MatchLabels and/or MatchExpressions are, Include will behave as if ['*'] were given. If
+                  MatchExpressions and MatchLabels are both not provided, Include must be provided to
```
I understand what you mean by "ANDed", I've never seen it written like that though. From my perspective, you are saying the the rules are combined by using the AND function. Let me your thoughts. I'm a little confused with if, then explanation. Are you saying that if parameter values are not set to "Include" then the parameter value is considered empty? 

Need to be more clear about "behave"; reason from style guide: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism

```suggestion
                  NamespaceSelector defines the list of namespaces to include or exclude for objects defined in
                  spec.objectTemplates. All selector rules are combined by using the AND function. If the parameter is not provided but
                  MatchLabels and/or MatchExpressions are, Include will behave as if ['*'] were given. If
                  MatchExpressions and MatchLabels are both not provided, Include must be provided to
```

dhaiducek replied:
I'll work on updating this--we have wording in the RHACM docs that I can probably pull to help clarify it (sorry, I must not have done that for this description).

dockerymick commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -37,71 +42,80 @@ spec:
-                    description: '''exclude'' is an array of filepath expressions
-                      to exclude objects by name.'
+                    description: Exclude is an array of filepath expressions to exclude
+                      objects by name.
```
continue review here

dockerymick commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -160,53 +177,77 @@ spec:
                             format: int64
                             type: integer
                           sans:
+                            description: Sans is a list of subject alternative names
+                              in the certificate.
```
Should "San" be displayed as an acronym? Or is "San" displayed like that from the policy?
```suggestion
                            description: SANS is a list of subject alternative names
                              in the certificate.
```

dhaiducek replied:
While you're right it is an acronym, unfortunately it was originally defined it in the code as `Sans`, so we should probably keep it like that to be consistent.

dockerymick commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -160,53 +177,77 @@ spec:
                             type: string
                         type: object
+                      description: |-
+                        NonCompliantCertificatesList reports a map of the details for each noncompliant certificate,
+                        where the key comes from the "certificate-name" label, "certmanager.k8s.io/certificate-name"
+                        label, or defaults to the name of the secret.
```
```suggestion
                        NonCompliantCertificatesList reports a map of the details for each non-compliant certificate,
                        where the key comes from the "certificate-name" label, "certmanager.k8s.io/certificate-name"
                        label, or defaults to the name of the secret.
```

dockerymick commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -160,53 +177,77 @@ spec:
                 type: object
               compliant:
-                description: Compliant, NonCompliant, UnknownCompliancy
+                description: ComplianceState reports the observed status resulting
+                  from the policy's definition(s).
```
```suggestion
                description: ComplianceState reports the observed status resulting
                  from the definitions of the policy'.
```

dhaiducek commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -160,53 +177,77 @@ spec:
-                              A Duration represents the elapsed time between two instants
-                              as an int64 nanosecond count. The representation limits the
-                              largest representable duration to approximately 290 years.
+                              Expiry is the time.Duration representation of the certificate's expiration from the time of the
+                              report.
```
Sort of, but one is a string type while the other is an explicit `time.Duration` type (I'm not entirely sure why they're both there...)

dhaiducek replied:
I'll match the wording recommended in the previous comment.

dhaiducek commented:
Updated! Thanks for the reviews, @dockerymick, @mprahl, @yiraeChristineKim, @JustinKuli!

@dockerymick Would you please verify that `noncompliant` should have a hyphen? We were trying to be consistent, so I've removed the hyphen elsewhere, and an internet search seems to indicate that it should not have a hyphen?

dockerymick commented:
@dhaiducek good call out! You are correct. Here is also a reference to confirm that the hyphen should not be used: https://www.ibm.com/docs/en/ibm-style?topic=grammar-prefixes

dockerymick commented on api/v1/certificatepolicy_types.go:
```diff
@@ -69,73 +66,122 @@ func (t Target) String() string {
+	// certificates. The Include and Exclude parameters accept file path expressions to include and
+	// exclude namespaces by name. The MatchExpressions and MatchLabels parameters specify namespaces
+	// to include by label. See the Kubernetes labels and selectors documentation. The resulting list
+	// is compiled by using the intersection of results from all parameters. Either Include or at
+	// least one of MatchExpressions or MatchLabels must be provide to retrieve namespaces.
```
```suggestion
	// is compiled by using the intersection of results from all parameters. You must provide either 
	// Include, MatchExpressions, or MatchLabels to retrieve namespaces.
```

dockerymick commented on deploy/crds/kustomize/policy.open-cluster-management.io_certificatepolicies.yaml:
```diff
@@ -37,71 +42,81 @@ spec:
+                  certificates. The Include and Exclude parameters accept file path expressions to include and
+                  exclude namespaces by name. The MatchExpressions and MatchLabels parameters specify namespaces
+                  to include by label. See the Kubernetes labels and selectors documentation. The resulting list
+                  is compiled by using the intersection of results from all parameters. Either Include or at
+                  least one of MatchExpressions or MatchLabels must be provide to retrieve namespaces.
```
```suggestion
                  is compiled by using the intersection of results from all parameters. You must provide either Include,
                  MatchExpressions, or MatchLabels to retrieve namespaces.
```

dockerymick commented on api/v1/certificatepolicy_types.go:
```diff
@@ -69,73 +66,122 @@ func (t Target) String() string {
 }
 
-// CertificatePolicySpec defines the desired state of CertificatePolicy
+// CertificatePolicySpec defines the certificates to be checked and the parameters within which
+// those certificates should fall.
```
Sorry for missing this the first time. Fall where? Do you mean the certificate policies will be removed? 
```suggestion
// CertificatePolicySpec defines the certificates to be checked and the parameters within which
// those certificates should fall.
```

dhaiducek replied:
As in the certificates targeted by the CertificatePolicy should be configured within the parameters set in the policy (otherwise they're noncompliant).

dockerymick replied:
Thanks Dale, but where do the policies go ? When I read "should fall", i interpreted 
```suggestion
// CertificatePolicySpec define which certificates need to be checked and the parameters within those policies
// are removed.
```

dhaiducek replied:
The cert-policy-controller doesn't take any actions--it only reports compliance or noncompliance of certificates by comparing certificates with the spec in the CertificatePolicy, and it's up to the user to take actions to fix the certificates.

dockerymick replied:
@dhaiducek thanks Dale, I made an edit to my suggestion. let me know your thoughts

dhaiducek replied:
Sure! Adding on to that, how about this:
```
// CertificatePolicySpec defines which certificates need to be checked and how those certificates
// should be configured to be compliant with the policy. Enforce remediation action is not
// currently supported, so you are responsible to take actions to fix any noncompliant certificates.
```

dhaiducek commented on api/v1/certificatepolicy_types.go:
```diff
@@ -20,34 +23,28 @@ const (
 )
 
-// ComplianceState shows the state of enforcement
+// ComplianceState reports the observed status resulting from the definitions of the policy.
```
Having "status results" feels repetitive to me. Would this be clearer?
```suggestion
// ComplianceState reports the observed status of the policy resulting from the definitions of the policy.
```

dockerymick replied:
Thanks for the suggestion, I want to remove the gerund "resulting"
```suggestion
// ComplianceState reports the observed status of the policy, which come from the definitions of the policy.
```

dhaiducek replied:
Okay, how about a similar phrasing:
```
// CertificatePolicyStatus reports the observed status that results from parameters of the
// certificate policy.
```

dockerymick replied:
I like that !

dockerymick commented on api/v1/certificatepolicy_types.go:
```diff
@@ -20,34 +23,28 @@ const (
 )
 
-// ComplianceState shows the state of enforcement
+// ComplianceState reports the observed status resulting from the definitions of the policy.
```
Using Dale's suggestion
```suggestion
// CertificatePolicyStatus reports the observed status that results from parameters of the
// certificate policy.
```

dhaiducek replied:
Since the `ComplianceState` field is directly under `CertificatePolicyStatus`, I'm thinking it could be tweaked a bit so that it's similar but not identical, something like this?
```
// ComplianceState reports the compliance that results from parameters of the certificate policy.
```

dockerymick reviewed (approved):
Thanks for your patience and hard work per usual @dhaiducek 
This looks good to me!

dhaiducek commented:
/unhold

JustinKuli reviewed (approved):
lgtm
